node/external/blockAPI/baseBlock.go

```diff
@@ -345,3 +354,164 @@ func bigIntToStr(value *big.Int) string {
+			attachTokensToAlteredAccount(apiAlteredAccount, altAccount, tokensFilter, withMetadata)
+		}
+
+		response.Accounts = append(response.Accounts, apiAlteredAccount)
+	}
+
+	return response
+}
+
+func attachTokensToAlteredAccount(apiAlteredAccount *common.AlteredAccountAPIResponse, altAccount *outport.AlteredAccount, tokensFilter string, withMetadata bool) {
+	for _, token := range altAccount.Tokens {
+		if !shouldAddTokenToResult(token.Identifier, tokensFilter) {
+			continue
+		}
+		if withMetadata {
+			apiAlteredAccount.Tokens = append(apiAlteredAccount.Tokens, token)
+			continue
+		}
+
+		apiAlteredAccount.Tokens = append(apiAlteredAccount.Tokens, &outport.AccountTokenData{
+			Identifier: token.Identifier,
+			Balance:    token.Balance,
+			Nonce:      token.Nonce,
+			Properties: token.Properties,
+			MetaData:   nil,
+		})
+	}
+}
+
+func shouldAddTokenToResult(tokenIdentifier string, tokensFilter string) bool {
+	if shouldIncludeAllTokens(tokensFilter) {
+		return true
+	}
+
+	return strings.Contains(tokensFilter, tokenIdentifier)
+}
+
+func shouldIncludeAllTokens(tokensFilter string) bool {
+	return tokensFilter == "*" || tokensFilter == "all"
+}
+
+func (bap *baseAPIBlockProcessor) apiBlockToTxsPool(apiBlock *api.Block) *outport.Pool {
+	pool := &outport.Pool{
+		Txs:     make(map[string]data.TransactionHandlerWithGasUsedAndFee),
+		Scrs:    make(map[string]data.TransactionHandlerWithGasUsedAndFee),
+		Invalid: make(map[string]data.TransactionHandlerWithGasUsedAndFee),
+		Rewards: make(map[string]data.TransactionHandlerWithGasUsedAndFee),
+		Logs:    make([]*data.LogData, 0),
+	}
+
+	for _, miniBlock := range apiBlock.MiniBlocks {
+		for _, tx := range miniBlock.Transactions {
+			bap.addTxToPool(tx, pool)
+			bap.addLogsToPool(tx, pool)
+		}
+	}
+
+	return pool
+}
+
+func (bap *baseAPIBlockProcessor) addLogsToPool(tx *transaction.ApiTransactionResult, pool *outport.Pool) {
+	if tx.Logs == nil {
+		return
+	}
+
+	logAddressBytes, err := bap.addressPubKeyConverter.Decode(tx.Logs.Address)
+	if err != nil {
+		log.Warn("altered accounts API: cannot decode log's address", "address", tx.Logs.Address, "error", err)
+		return
+	}
+
+	logsEvents := make([]*transaction.Event, 0)
+	for _, logEvent := range tx.Logs.Events {
+		eventAddressBytes, err := bap.addressPubKeyConverter.Decode(logEvent.Address)
+		if err != nil {
+			log.Warn("altered accounts API: cannot decode event's address", "address", tx.Logs.Address, "error", err)
+			continue
+		}
+
+		logsEvents = append(logsEvents, &transaction.Event{
+			Address:    eventAddressBytes,
+			Identifier: []byte(logEvent.Identifier),
+			Topics:     logEvent.Topics,
+			Data:       logEvent.Data,
+		})
+	}
+
+	pool.Logs = append(pool.Logs, &data.LogData{
+		LogHandler: &transaction.Log{
+			Address: logAddressBytes,
+			Events:  logsEvents,
+		},
+		TxHash: tx.Hash,
+	})
+}
+
+func (bap *baseAPIBlockProcessor) addTxToPool(tx *transaction.ApiTransactionResult, pool *outport.Pool) {
```

Comment:
Right in this case, but we could have a `TODO` to actually check what is the needed input for `AlteredAccountsProvider`. 
In case it only needs a list of addresses, there is no need to pass such a large struct (tx pool) instead of a simple `[]addresses`

Reply:
it is not a large struct. it only contains pointers to transactions that only have 2 fields. Added a TODO however